### PR TITLE
dark mode

### LINKED ISSUE #25: Dark Mode
New option similar to handout/recording that enables slides in dark mode. I would not suggest people to use it for live presentations, but still allow any combination of recording/handout and dark mode. This could be useful for pure online presentations.

### PR DESCRIPTION
**Pull request by EagleoutIce**
Closes #25.

Open for discussion!

### CODE REVIEW

File: beamerthemeuulm.sty [hunk 1]
```diff
@@ -55,14 +57,38 @@
...
+\setbeamercolor{frametitle}{fg=uulm@text}
+\setbeamercolor{structure}{fg=uulm@text}
+\setbeamercolor{normal text}{fg=uulm@text}
+\setbeamercolor*{palette primary}{use=structure,fg=structure.fg}
+\setbeamercolor*{palette secondary}{use=structure,fg=structure.fg!95!black}
+\setbeamercolor*{palette tertiary}{use=structure,fg=structure.fg!90!black}
+\setbeamercolor*{palette quaternary}{use=structure,fg=structure.fg!95!black,bg=black!80}
+\setbeamercolor{item}{fg=uulm@text}
+\setbeamercolor{myfooter}{fg=uulm@text,bg=darkgray!90!black}%
+% overwrite faculty?
```
[the remark this answers is not in the document]
Comment (EagleoutIce) [reply beamerthemeuulm.sty:74]: The faculty colors are still available, the comment is more meant as a TODO regarding:
maybe we want a different variant of these colors when using the darkmode (e.g., a lighter red)

File: beamerthemeuulm.sty [hunk 2]
```diff
@@ -55,14 +57,38 @@
...
+\AtEndPreamble{\setbeamercolor{mypagenumber}{fg=uulm@text,bg=darkgray!40!black}}
+\setbeamercolor{logobox}{bg=uulm@background}
```
[the remark this answers is not in the document]
Comment (EagleoutIce) [reply beamerthemeuulm.sty]: I adapted it to be the font color (different, lighter grays looked wrong)

File: beamerthemeuulm.sty [hunk 3]
```diff
@@ -556,19 +581,19 @@
...
         left=#4, right=#4, top=#4, bottom=#4, bottomtitle=-.5mm,
         #1
     },
     % style definition for the normal boxes
     % #1: extra arguments, #2: title, #3: color of the frame and background
-    uulm@boxstyle/.style n args=3{uulm@basic-boxstyle={#1}{#2}{#3}{1mm}, colback=#3!10},
+    uulm@boxstyle/.style n args=3{uulm@basic-boxstyle={#1}{#2}{#3}{1mm}, colback=#3!10!uulm@background},
     % style definition for their tight variant
     % #1: extra arguments, #2: title, #3: color of the frame
-    uulm@tightboxstyle/.style n args=3{uulm@basic-boxstyle={#1}{#2}{#3}{0mm}, colback=white}
```
[the remark this answers is not in the document]
Comment (EagleoutIce) [reply beamerthemeuulm.sty:571]: So we keep it for now (but we can easily change it later)

**Comment by tthuem** [comment]
@EagleoutIce I have extensively tested the dark theme and made slight modifications in my last commit: [f604994](https://github.com/SoftVarE-Group/SlideTemplate/pull/54/commits/f6049941ed107bdb3504d8b8710cb06da6c76360)

In particular, it was not quite easy to find settings such that colored text (e.g., blue/red/green/orange) is still readable on all different backgrounds (regular background, title of a colored box, area within a colored box). Here are some examples:

https://github.com/SoftVarE-Group/Course-on-Software-Product-Lines/raw/main/slides/01-introduction.pdf
https://github.com/SoftVarE-Group/Course-on-Software-Product-Lines/raw/main/slides/02-runtime.pdf
https://github.com/SoftVarE-Group/Course-on-Software-Product-Lines/raw/main/slides/03-cloneandown.pdf
https://github.com/SoftVarE-Group/Course-on-Software-Product-Lines/raw/main/slides/04-modeling.pdf

For me, the dark mode is ready to go into the release branch.

**Comment by tthuem** [comment]
@Benno2000 Can you merge the PR?

**Comment by Benno2000** [comment]
The changes look good to me, too! Now everything remains readable also in darkmode. I will merge this PR now.

**Pull request merged**